src/tokens/Datatoken.ts

```diff
@@ -973,13 +975,73 @@ export class Datatoken {
       new this.web3.eth.Contract(this.datatokensAbi, dtAddress),
       this.config
     )
+
     if (!consumeMarketFee) {
       consumeMarketFee = {
         consumeMarketFeeAddress: '0x0000000000000000000000000000000000000000',
         consumeMarketFeeToken: '0x0000000000000000000000000000000000000000',
         consumeMarketFeeAmount: '0'
       }
     }
+
+    const publishMarketFee = await dtContract.methods.getPublishingMarketFee().call()
+    const tokens = [
+      {
+        token: providerFees.providerFeeToken,
+        feeAmount: providerFees.providerFeeAmount
+      },
+      {
+        token: consumeMarketFee.consumeMarketFeeToken,
+        feeAmount: parseFloat(consumeMarketFee.consumeMarketFeeAmount)
+      },
+      {
+        token: publishMarketFee[1],
+        feeAmount: parseFloat(publishMarketFee[2])
+      }
+    ]
+
+    const uniqueTokens = []
+    tokens.map((address) => {
+      if (uniqueTokens.length > 0) {
+        uniqueTokens.map((uAddress) => {
+          if (uAddress.token === address.token) {
+            uAddress.feeAmount += address.feeAmount
+          } else {
+            uniqueTokens.push({
+              token: address.token,
+              feeAmount: address.feeAmount
+            })
+          }
+        })
+      } else {
+        uniqueTokens.push({
+          token: address.token,
+          feeAmount: address.feeAmount
+        })
+      }
+    })
+
+    const getCurrentAllownceTokens = uniqueTokens.map(async (token) => {
+      if (token.token === ZERO_ADDRESS || token.feeAmount === 0) return token
+      const currentAllowance = await allowance(this.web3, token.token, address, consumer)
+      if (
+        new Decimal(currentAllowance).greaterThanOrEqualTo(new Decimal(token.feeAmount))
+      ) {
+        LoggerInstance.error(`ERROR: Failed checking allowance: ${token.token}`)
+        throw new Error(`allowance (${currentAllowance}) is too low`)
+      } else {
+        token.currentAllowance = currentAllowance
+        return token
+      }
+    })
+
+    try {
+      const allownceTokens = await Promise.all(getCurrentAllownceTokens)
+    } catch (e) {
+      LoggerInstance.error(`ERROR: Failed checking allowance : ${e}`)
+      throw new Error(`Failed checking allowance: ${e}`)
+    }
```

Comment:
Think moving this to a another method maybe under utils would be better and just calling the method here, since it's adding a lot of code to the startOrder method alters method's scope

Reply:
got it 👍🏽 you're right. I'll move it into utils as it makes more sense there. Thanks